src/nifreeze/model/pet.py

```diff
@@ -95,62 +125,114 @@ def __init__(
         if self._x[-1] > (self._xlim - DEFAULT_TIMEFRAME_MIDPOINT_TOL):
             raise ValueError("Last frame midpoint should not be equal or greater than duration")
 
-        # Calculate index coordinates in the B-Spline grid
-        self._n_ctrl = n_ctrl or (len(timepoints) // 4) + 1
+    def _preproces_data(self) -> np.ndarray:
```

Comment:
preprocess?

Reply:
Resolved in b2898e2. I will squash once the rest gets resolved.